web-bundle/src/main/java/com/graphhopper/replica/GtfsLinkMapper.java

```diff
@@ -115,89 +114,86 @@ public void setGtfsLinkMappings() {
                     .filter(stop -> stopIdsForStreetBasedTrips.contains(stop.stop_id))
                     .collect(Collectors.toMap(stop -> stop.stop_id, stop -> stop));
 
+            // We only care to track the unique stop->stop pairs for each route (ignoring trips).
+            SetMultimap<String, Pair<Stop, Stop>> routeIdToStopPairs = HashMultimap.create();
+            tripIdToStopsInTrip.keySet().stream()
+                .forEach(tripId -> {
+                    getODStopsForTrip(tripIdToStopsInTrip.get(tripId), stopsForStreetBasedTrips).stream()
+                        .forEach(stopPair -> {
+                            routeIdToStopPairs.put(feed.trips.get(tripId).route_id, stopPair);
+                        });
+                });
+            Set<Pair<Stop, Stop>> uniqueStopPairs = routeIdToStopPairs.values().stream().collect(Collectors.toSet());
```

Comment:
take it or leave it nit: IntelliJ suggested `Sets.newHashSet(routeIdToStopPairs.values());` here instead of using `stream`, might be slightly simpler :)